### DISCUSSION TITLE
anime4k如何用标准的opengl实现啊

### DISCUSSION THREAD

**Discussion by ZadeAo** [Blinue/Magpie#618] [Dev Channel]
我使用framebuffer作为保存各层卷积层的输出。原始的glsl文件中的#define go_1(x_off, y_off) (max(-(conv2d_tf_texOff(vec2(x_off, y_off))), 0.0))。go_1 被我翻译成max(-texture2D(tex0, vec2(tc.x+offsetX*imageWidthFactor,tc.y+offsetY*imageHeightFactor)),0.0)，结果明显永远返回是黑色，因为texture2D返回的0-1范围的4维向量，求正确做法是啥

**Comment by Blinue** [comment]
我猜是纹理格式问题，我们使用的是 R16G16B16A16_FLOAT。

**Reply by ZadeAo** [reply]
好的，我试试

**Reply by ZadeAo** [reply]
有效果，但和桌面端比有差距，还是要研究下怎么回事
